azure-iot-device/azure/iot/device/iothub/aio/async_inbox.py

```diff
@@ -4,10 +4,24 @@
 # license information.
 # --------------------------------------------------------------------------
 """This module contains an Inbox class for use with an asynchronous client"""
-
+import asyncio
+import threading
 import janus
 from azure.iot.device.iothub.sync_inbox import AbstractInbox
 
+# This logic could potentially be encapsulated in another module...
+# Say... some kind of... loop_manager.py?
+INBOX_LOOP = asyncio.new_event_loop()
+INBOX_THREAD = threading.Thread(target=INBOX_LOOP.run_forever)
+INBOX_THREAD.daemon = True
+INBOX_THREAD.start()
+
+# IMPLEMENTATION NOTE: The janus Queue exists entirely on the above mentioned INBOX_LOOP,
```

Comment:
>The janus Queue exists entirely on the above mentioned INBOX_LOOP [](start = 23, length = 65)

Nice hack.  Like you say, not pretty, but it's simple and it's the right size solution.  Adding a loop manager would be overkill IMO.

Reply:
That being said, the more I think about this, the more nervous I become.  I like the idea of an INBOX_LOOP, but I don't like the thread being created as a side-effect of importing the module.  Can't you wait to crate the thread/loop until you actually instantiate something?  Also, if INBOX_THREAD crashes (or otherwise stops running), can the application detect this and recover?  If INBOX_LOOP is a global like this, the only recovery is a reboot.  I'm not sure the app can even detect that this needs to happen though.

---
In reply to: [473995318](https://github.com/Azure/azure-iot-sdk-python/pull/644#discussion_r473995318) [](ancestors = 473995318)